lib/membrane_protocol_icecast/output/controller.ex

```diff
@@ -0,0 +1,35 @@
+defmodule Membrane.Protocol.Icecast.Output.Controller do
+  @type state_t :: any
+
+  @type invalid_reason_t ::
+    {:request, {:request, any} | {:header, binary}} |
+    {:method, atom | charlist} |
+    {:mount, binary} |
+    :format_unknown |
+    :format_not_allowed |
+    :request |
+    :too_many_headers |
+    :unauthorized
+    # FIXME method -> method_*
+
+  @type incoming_reply_t :: {:ok, {:allow, any} | {:deny, :forbidden}}
+  @type listener_reply_t :: {:ok, {:allow, any}, Types.format_t} | {:ok, {:deny, :forbidden, :not_found}}
+
+  # Initial actions
+  @callback handle_init(any) :: {:ok, state_t}
```

Comment:
Specify types for options

Reply:
This argument can be of type `any`. It's up to the user to decide what is passed here.